src/games/strategy/triplea/ui/ErrorHandler.java

```diff
@@ -1,26 +1,47 @@
 package games.strategy.triplea.ui;
 
-import games.strategy.engine.GameOverException;
+import games.strategy.debug.ClientLogger;
 
 /**
- * NOt entirly safe or elegant.
- * We want to ignore game over exceptions when the game is actually over.
- * This assumes only 1 game in a vm at a time.
+ * When dealing with swing threads and new threads, exception handling can get tricky. Namely without
+ * a handler to catch exceptions in these new threads, the stack traces will be poor. Specifically you
+ * will get a stack trace that points to where you started the thread and not to the actual line within
+ * the thread that had the problem.
+ *
+ * To solve this unhandled exception handlers get registered. For more details, see:
+ * http://stackoverflow.com/questions/75218/how-can-i-detect-when-an-exceptions-been-thrown-globally-in-java#75439
  */
-public class ErrorHandler {
-  private static volatile boolean m_isGameOver;
+public class ErrorHandler implements Thread.UncaughtExceptionHandler, ErrorHandlerAwtEvents {
 
-  public static void setGameOver(final boolean aBool) {
-    m_isGameOver = aBool;
+  @Override
+  public void uncaughtException(Thread t, Throwable e) {
+    handle(e);
   }
 
-  public ErrorHandler() {}
-
-  public void handle(final Throwable t) {
-    if (t instanceof GameOverException && m_isGameOver) {
```

Comment:
This is quite poor design and has other problems. Two design (code smells) I can spot: first the state of "game over" is not communicated well throughout the system, it's a set of static and not static calls rather than something more robust and decoupled like an event listener pattern. Secondly, by having some modules check for game over and then throw this exception in order to abort code logic, that is control-flow by exception handling, which is also a design smell.

In addition of important note, this code buries exceptions rather than ensuring we deal with them. We do not know the velocity of this exception, nor if we successfully told every part of the code that the game is over and resources can be released. If we missed something, it will be ignored here.

Last, from an abstraction and responsibility point of view, it doesn't make sense to tell the *ErrorHandler* anything about the game state. No errors should be thrown before, during, or after a game. If so, we use this class to *tell* us about it so we can fix it. Ignoring cleanup errors, or who knows whatever kind of error simply because the game is over does not mean the error has gone away, will go away, or does not represent a problem we would want to know about.

Hopefully you'll agree that is a very compelling class to remove the "game over" concept and associated responsibilities from this class.